System.IO.FileSystem/Directory.cs

```diff
@@ -45,6 +45,10 @@ public static bool CreateDirectory(string path)
         /// <exception cref="IOException">This method will throw DirectoryNotEmpty exception if folder is not empty.</exception>
         public static void Delete(string path, bool recursive = false)
         {
+            if (string.IsNullOrEmpty(path))

```

Comment:
These checks for NULL args are, of course, important, but they are more efficiently implemented at the native end.

On methods that are passing these to a native call, the check should be done there.
The exception being thrown is ArgumentException, so the Intellisense comment should reflect this, instead of IOException.

The same goes for the similar ones at the following methods.

Reply:
> 
> 
> These checks for NULL args are, of course, important, but they are more efficiently implemented at the native end.
> 
> On methods that are passing these to a native call, the check should be done there.
> The exception being thrown is ArgumentException, so the Intellisense comment should reflect this, instead of IOException.
> 
> The same goes for the similar ones at the following methods.

I reverted the checks. I added checks based on File.cs, but i missed the point that for now methods are only calling native methods, so the checks are in fact not needed.
Exceptions should now be correct when, the checks are removed.

Additionally CreateDirectory implementation differs from original [API ](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=net-5.0), as it would require to implement DirectoryInfo and FileSystemInfo classes. I don't have time for this task right now, and also its added value is almost none.

Reply:
> I reverted the checks. I added checks based on File.cs, but i missed the point that for now methods are only calling native methods, so the checks are in fact not needed.

Those checks are needed, but they are better performed at native end.


> Additionally CreateDirectory implementation differs from original [API ](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=net-5.0), as it would require to implement DirectoryInfo and FileSystemInfo classes. I don't have time for this task right now, and also its added value is almost none.

Agreed on this. For the time being we don't seem to need DirectoryInfo

Reply:
@josesimoes Do you have any more suggestions for improvements, or maybe something more needs to be implemented?
I will need to test the firmware, if it is working correctly. If it does i think those two PR could be ready to merge.

Reply:
Nothing else I can think of. Please go ahead with the testing and report back. 👍🏻

Reply:
@josesimoes I tested every method implemented by System.IO on MIMXRT1060 microcontroller. Fixed problems which showed up. 
It should work on other architectures, but i suppose that we will know for sure, only when people will start migrating from Windows.Storage to System.IO api.

Reply:
Yeap! Let's go for this.